packages/e2e-tests/specs/undo.test.js

```diff
@@ -63,4 +65,36 @@ describe( 'undo', () => {
 		// After undoing every action, there should be no more undo history.
 		expect( await page.$( '.editor-history__undo[aria-disabled="true"]' ) ).not.toBeNull();
 	} );
+
+	it( 'should undo for explicit persistence editing post', async () => {
+		// Regression test: An issue had occurred where the creation of an
+		// explicit undo level would interfere with blocks values being synced
+		// correctly to the block editor.
+		//
+		// See: https://github.com/WordPress/gutenberg/issues/14950
+
+		// Issue is demonstrated from an edited post: create, save, and reload.
+		await clickBlockAppender();
+		await page.keyboard.type( 'before' );
+		await pressKeyWithModifier( 'primary', 's' );
```

Comment:
Possible issue here being that we don't actually wait for the save to happen. I'm going to swap this out with using `createNewPost` to provide the content.